src/v/cluster/rm_stm.cc

```diff
@@ -65,17 +65,19 @@ static model::record_batch make_fence_batch(model::producer_identity pid) {
 static model::record_batch make_fence_batch(
   model::producer_identity pid,
   model::tx_seq tx_seq,
-  std::chrono::milliseconds transaction_timeout_ms) {
+  std::chrono::milliseconds transaction_timeout_ms,
+  model::partition_id tm) {
     iobuf key;
     auto pid_id = pid.id;
     reflection::serialize(key, model::record_batch_type::tx_fence, pid_id);
 
     iobuf value;
     reflection::serialize(
       value,
-      rm_stm::fence_control_record_version,
+      rm_stm::fence_control_record_tm_version,
```

Comment:
```
static constexpr int8_t fence_control_record_v0_version{0};
static constexpr int8_t fence_control_record_version{1};
```

fence_control_record_version points always to last version

you should update

```
static constexpr int8_t fence_control_record_v0_version{0};
static constexpr int8_t fence_control_record_v1_version{1};
static constexpr int8_t fence_control_record_version{2};
```

instead of introducing `fence_control_record_tm_version`

Reply:
fixed